### PR TITLE
Allow admin to hide and move navigation menu items

### PR DESCRIPTION
**Pull request by foxerest**
### Summary of changes:

- Add "move to" button to change an item's parent;
- Add dnd to change the order of navigation menu items;
- Add "show" and "hide" buttons to change an item's visibility;

[User Story Requirements SPHUB-55](https://github.com/dark-side/lanthanum/tree/master/sports_hub_portal/web_application_features/maintain_navigation/user_stories/hide_show_navigation_items)
[User Story Requirements SPHUB-56](https://github.com/dark-side/lanthanum/tree/master/sports_hub_portal/web_application_features/maintain_navigation/user_stories/move_and_order_navigation_items)

### CODE REVIEW

File: src/components/AdminContextAreaManagement/InformationArchitecture/IAComponents/NavigationMenuItem/NavigationMenuItem.js [hunk 1]
```diff
@@ -24,59 +40,199 @@ export default function NavigationMenuItem({name, itemType, isActive, onPress})
...
+                    item.name}
+            </span>
+                <div className="dropDown-wrapper" ref={menuRef}>
+                    {isItemHovering && !
+                        isStartDrag && (
+                        <button className="item-menu-button" data-testid = "menu-button" onClick={() => setIsMenuOpen(!isMenuOpen)}>
                             <IconContext.Provider value={{size: "20", className: "menu-icon"}}>
-                                <BsThreeDots style={{color: isMenuOpen ? "#D72130" : "#B2B2B2"}}/>
+                                <BsThreeDots style={{color:
+                                        isStartDrag ? "#F4C7CA": isMenuOpen ? "#D72130" : "#B2B2B2"}}/>
```
Comment (Yurii-Hubych) [inline src/components/AdminContextAreaManagement/InformationArchitecture/IAComponents/NavigationMenuItem/NavigationMenuItem.js]: Could you please, undo these changes

File: src/components/AdminContextAreaManagement/InformationArchitecture/IAComponents/NavigationMenuItem/NavigationMenuItem.js [hunk 2]
```diff
@@ -24,59 +40,199 @@ export default function NavigationMenuItem({name, itemType, isActive, onPress})
...
+                                            color: (isMoveToHovering||isCategoriesOpen)  ? "#D72130" : "#B2B2B2"}}/>
+                                    </IconContext.Provider>
+                                </div>
+                            )}
+                            {isCategoriesOpen && (
+                                <div className="dropDown-categories">
+                                    {
+                                        categories.filter(item => filterCategories(item,
+                                            item.categoryId))
+                                        .map((c, index) =>
```
Comment (Yurii-Hubych) [inline src/components/AdminContextAreaManagement/InformationArchitecture/IAComponents/NavigationMenuItem/NavigationMenuItem.js]: Could you please, do this in one line

File: src/components/AdminContextAreaManagement/InformationArchitecture/IAComponents/NavigationMenuItem/NavigationMenuItem.js [hunk 3]
```diff
@@ -24,59 +40,199 @@ export default function NavigationMenuItem({name, itemType, isActive, onPress})
...
+                                             onMouseOver={() => setActiveHoverCategory(index)}
+                                             onMouseOut={() => setActiveHoverCategory(-1)}
+                                             onClick={(e) => handleCategoryClick(e, c.id)}>
+                                            <h2 className="items-name">{c.name}</h2>
+                                            {
+                                                itemType === "team" && (
+                                                <IconContext.Provider value={{size: "14", className: "move-to-icon"}}>
+                                                    <GoTriangleRight
+                                                        style={{color: (activeHoverCategory === index ||
+                                                                activeClickCategory === c.id) ? "#D72130" : "#B2B2B2"}}/>
```
Comment (Yurii-Hubych) [inline src/components/AdminContextAreaManagement/InformationArchitecture/IAComponents/NavigationMenuItem/NavigationMenuItem.js]: Could you please, do this in one line

File: src/components/AdminContextAreaManagement/InformationArchitecture/IAComponents/NavigationMenuItem/NavigationMenuItem.test.js [hunk 4]
```diff
@@ -4,14 +4,17 @@ import { render, screen, fireEvent } from "@testing-library/react";
 import NavigationMenuItem from "./NavigationMenuItem";
 
 test("Navigation menu item should be rendered", () => {
+    let item = {name: "name", isHidden: false}
+
     render(
-        <NavigationMenuItem name="testName" isActive={false} itemType="subcategory" onPress={() => void(0)}/>
+        <NavigationMenuItem name="testName" item={item} isActive={false} changeDraggable = {() => void(0)}
+                            itemType="subcategory" onPress={() => void(0)}/>
```
Comment (Yurii-Hubych) [inline src/components/AdminContextAreaManagement/InformationArchitecture/IAComponents/NavigationMenuItem/NavigationMenuItem.test.js:11]: Could you please, do it in one line

File: src/components/AdminContextAreaManagement/InformationArchitecture/IAComponents/NavigationMenuItem/NavigationMenuItem.test.js [hunk 5]
```diff
@@ -20,8 +23,11 @@ test("Navigation menu item should be rendered", () => {
 });
 
 test("Dropdown menu should be rendered", () => {
+    let item = {name: "name", isHidden: false}
+
     render(
-        <NavigationMenuItem name="testName" isActive={false} itemType="subcategory" onPress={() => void(0)}/>
+        <NavigationMenuItem name="testName" item={item} isActive={false}
+                            changeDraggable = {() => void(0)} itemType="subcategory" onPress={() => void(0)}/>
```
Comment (Yurii-Hubych) [inline src/components/AdminContextAreaManagement/InformationArchitecture/IAComponents/NavigationMenuItem/NavigationMenuItem.test.js]: Could you please, do it in one line

File: src/services/TeamService/TeamService.js [hunk 6]
```diff
@@ -18,19 +18,41 @@ export default class TeamService {
...
     }
 
-    createTeam(data) {
+    createTeam(name, subcategoryId) {
         const method = 'POST';
-        return this.request(urls.teamsEndpoint, method, data)
+
+        let formData = new FormData();
+        formData.append('name', name);
+        formData.append('subcategoryId', subcategoryId);
```
Comment (Yurii-Hubych) [inline src/services/TeamService/TeamService.js]: Could you please, move this to your component

File: src/services/TeamService/TeamService.js [hunk 7]
```diff
@@ -18,19 +18,41 @@ export default class TeamService {
...
+
+    editTeam(data) {
+        const method = 'PUT';
+
+        let formData = new FormData();
+        formData.append('id', data.id);
+        formData.append('name', data.name);
+        formData.append('subcategoryId', data.subcategoryId);
+        formData.append('isHidden', data.isHidden);
+        formData.append('OrderIndex', data.orderIndex);
```
Comment (Yurii-Hubych) [inline src/services/TeamService/TeamService.js]: Could you please, move this to your component

**Review by Yurii-Hubych — CHANGES_REQUESTED** [review]
Everything else looks good to me 😄

**Review by nadiiakhasyshyn — APPROVED** [review]

**Review by Yurii-Hubych — APPROVED** [review]

**Review by markoparfeniuk — APPROVED** [review]
